rustls-libssl/src/cache.rs

```diff
@@ -235,8 +235,20 @@ impl ServerSessionStorage {
     fn insert(&self, new: Arc<SslSession>) -> bool {
         self.tick();
 
+        let max_size = self
+            .parameters
+            .lock()
+            .map(|inner| inner.max_size)
+            .unwrap_or_default();
+
         if let Ok(mut items) = self.items.lock() {
-            items.insert(new)
+            let inserted = items.insert(new);
+
+            while items.len() > max_size {
+                Self::flush_oldest(&mut items);
+            }
```

Comment:
I think because `0` as max implies "no maximum" this should be something like:

```

            while items.len() > max_size && max_size > 0 {
                Self::flush_oldest(&mut items);
            }
```

Reply:
Good point, I have overlooked the size == 0 case. I will add a separate commit that does that (also for the client cache)